packages/dapp/src/components/FeatureCards/FeatureCard.tsx

```diff
@@ -72,6 +80,21 @@ export const FeatureCard = ({ data, isSuccess, assets }: FeatureCardProps) => {
               !data?.fields?.displayConditions?.hasOwnProperty('showOnce') &&
                 !!data?.fields?.displayConditions?.id &&
                 onDisableFeatureCard(data?.fields?.displayConditions?.id);
+              trackEvent({
+                category: TrackingCategories.FeatureCard,
+                action: TrackingActions.CloseFeatureCard,
+                label: `close_${data?.fields?.displayConditions?.id}`,
+                data: {
+                  [TrackingEventParameters.FeatureCardTitle]:
+                    data?.fields?.title,
+                  [TrackingEventParameters.FeatureCardId]:
+                    data?.fields?.displayConditions?.id,
+                },
+                disableTrackingTool: [
+                  EventTrackingTool.ARCx,
+                  EventTrackingTool.Raleon,
+                ],
+              });
```

Comment:
Can we please extract these inline functions to some handlers at the top of the component? It becomes hard to maintain and generally considered a bad practice 🙂

Reply:
Good point! Thank you, I found a few of those and refactored